### DISCUSSION TITLE
[mobx-state-tree] Error while converting {"config":{"assembly":...} to ReactLinearGenomeView

### DISCUSSION THREAD

**Discussion by fromYuechenimportQi** [GMOD/jbrowse-components#3529] [Q&A] [answered]
Hello, I wanted to **embed** Jbrowse2 in my website.

1.  I followed your tutor in [https://jbrowse.org/jb2/docs/tutorials/embed_linear_genome_view/](url), everything went well.

2.  I changed **assembly** and **track** file in the tutor to my own **genome fasta** and **gff3**, and revised assembly.js and tracks.js

3. After using `npx serve` and opening it in the browser, there was an error `[mobx-state-tree] Error while converting {"config":{"assembly":{"name":"Zea_mays","sequence":{"type":"Zea_maysReferenceSequenceTrack","trackId":"MyReferenceSequenceTrack","adapter":{"type":"BgzipFastaAdapter","fastaLocation":{"uri":"Zea_mays.fa.gz","locationType":"UriLocation"},"faiLocation"......iew"}}} to ReactLinearGenomeView`

4.`try{} catch{}` was added, and I found that the error comes from `const state = new createViewState({assembly,tracks})`

5. If `tracks` in the `const state = new createViewState({assembly,tracks})` was deleted, the error never appeared again

**Comment by cmdcolin** [comment, marked as answer]
can you paste your assembly and tracks? probably tracks is the relevant one. the config schema checking is quite strict so it would probably be due to an error related to the track configs

**Reply by cmdcolin** [reply]
sure thing, I meant if you had the config you used though :) this is an example config with a tabix gff3, some examples here https://jbrowse.org/jb2/docs/tutorials/embed_linear_genome_view/04_tracks/#creating-track-configurations-with-the-cli

**Reply by fromYuechenimportQi** [reply]
Thank you again :)
Tracks.js: 
```
export default {
    "type": "FeatureTrack",
    "trackId": "Zea_mays",
    "name": "Zea_mays.sorted.gff",
    "Category": ['Genes'],
    "adapter": {
        "type": "Gff3TabixAdapter",
        "gffGzLocation": {
            "uri": "Zea_mays.sorted.gff.gz",
            "locationType": "UriLocation"
        },
        "index": {
            "location": {
                "uri": "Zea_mays.sorted.gff.gz.tbi",
                "locationType": "UriLocation"
            },
            "indexType": "TBI"
        }
    },
    "renderer": {
        "type": 'SvgFeatureRenderer',
    },
    "assemblyNames": ["Zea_mays"]
}
```
Assembly.js:
```
export default {
    "name": "Zea_mays",
    "sequence": {
        "type": "ReferenceSequenceTrack",
        "trackId": "Zea_mays-ReferenceSequenceTrack",
        "adapter": {
            "type": "BgzipFastaAdapter",
            "fastaLocation": {
                "uri": "Zea_mays.fa.gz",
                "locationType": "UriLocation"
            },
            "faiLocation": {
                "uri": "Zea_mays.fa.gz.fai",
                "locationType": "UriLocation"
            },
            "gziLocation": {
                "uri": 'Zea_mays.fa.gz.gzi',
                "locationType": 'UriLocation',
            }
        }
    }
}
```

**Reply by cmdcolin** [reply]
@yueyue970506 I think this error is due to your tracks.js exporting a single track object instead of an array, should be able to change to

```
export default [{
    "type": "FeatureTrack",
    "trackId": "Zea_mays",
    "name": "Zea_mays.sorted.gff",
    "Category": ['Genes'],
    "adapter": {
        "type": "Gff3TabixAdapter",
        "gffGzLocation": {
            "uri": "Zea_mays.sorted.gff.gz",
            "locationType": "UriLocation"
        },
        "index": {
            "location": {
                "uri": "Zea_mays.sorted.gff.gz.tbi",
                "locationType": "UriLocation"
            },
            "indexType": "TBI"
        }
    },
    "renderer": {
        "type": 'SvgFeatureRenderer',
    },
    "assemblyNames": ["Zea_mays"]
}]
```

**Reply by fromYuechenimportQi** [reply]
It works!!!!!
Thank you very much!!!
